### ISSUE TITLE
Error printing for-loop scoping deprecation message

### ISSUE THREAD

**Issue by mbauman** [JuliaLang/julia#27870]
Didn't this use to print a nice warning or error about the need for a `global i` inside the for loop body?

```julia
julia> i = 1; for _ in 1:1; i += i; end
Exception handling log message: MethodError(print, (IOBuffer(data=UInt8[...], readable=true, writable=true, seekable=true, append=false, size=2, maxsize=Inf, ptr=3, mark=-1), nothing), 0xffffffffffffffff)
```

**Comment by vtjnash** [comment]
Why is that throwing a MethodError? Are we miscompiling something? `print` should have at least some definition for all types.

**Comment by yuyichao** [comment]
> print should have at least some definition for all types.

Not anymore after https://github.com/JuliaLang/julia/pull/27829 .....

**Comment by JeffBezanson** [comment]
It now throws a MethodError on `nothing`. We could change that to a different kind of error if we want though.

**Comment by JeffBezanson** [comment]
What I really want to know is why some logging code is trying to print `nothing`.

**Comment by fredrikekre** [comment]
> What I really want to know is why some logging code is trying to print `nothing`.

The module argument sometimes becomes `nothing`, I guess when it can't be determined for some reason. Incidentally I opened #27792 before `print(nothing)` was removed. Perhaps that will help a bit here at least? Perhaps it might also be that the file argument becomes `nothing`?

**Comment by JeffBezanson** [comment]
Fixed by #27792
